invenio_app_rdm/records_ui/views/decorators.py

```diff
@@ -403,3 +403,24 @@ def view(**kwargs):
         return view
 
     return decorator
+
+
+def no_cache_response(f):
+    """Add appropriate response headers to force no caching.
+
+    This decorator is used to prevent caching of the response in the browser. This is needed
+    in the deposit form as we initialize the form with the record metadata included in the html page
+    and we don't want the browser to cache this page so that the user always gets the latest version of the record.
+    """
+
+    @wraps(f)
+    def view(*args, **kwargs):
+        response = make_response(f(*args, **kwargs))
```

Comment:
Minor: I would check if the func already returns a Response, in that case we don't need the make_response, so it can be used by all funcs.
```suggestion
        result = f(*args, **kwargs)

        if isinstance(result, Response):
            response = result
        else:
            response = make_response(result)```